File: session-02-dependency-injection/typescript/src/main.ts
import {NestFactory} from "@nestjs/core";
import {AppModule} from "./app.module";

async function bootstrap() {
    const app = await NestFactory.create(AppModule);
    await app.listen(8080);
    console.log("Server running on http://localhost:8080");
    console.log("Try: curl http://localhost:8080/orders");
}

bootstrap();

File: session-02-dependency-injection/go/main.go
package main

import (
	"fmt"

	"coffeeshop-di/order"

	"github.com/gin-gonic/gin"
)

func main() {
	// This IS dependency injection in Go. No framework, no annotations, no magic.
	// You create the dependencies and pass them to constructors. That's it.
	repo := order.NewInMemoryOrderRepository()
	service := order.NewOrderService(repo)
	handler := order.NewHandler(service)

	r := gin.Default()
	r.GET("/orders", handler.ListOrders)
	r.GET("/orders/:id", handler.GetOrder)

	fmt.Println("Server running on http://localhost:8080")
	fmt.Println("Try: curl http://localhost:8080/orders")
	r.Run(":8080")
}
